road/scatter.py

```diff
@@ -0,0 +1,236 @@
+                    "align": "left",
+                    "baseline": "bottom",
+                    "dx": 5,
+                    "dy": -5
+                },
+                "data": {"values": [{"text": equation_text}]},
+                "encoding": {
+                    "x": {"value": 0},
+                    "y": {"value": 0},
+                    "text": {"field": "text", "type": "nominal"},
+                    "color": {"value": "red"}
+                }
+            },
+            {
+                "mark": "line",
+                "data": {"values": diagonal_df.to_dict(orient='records')},
+                "encoding": {
+                    "x": {"field": x_field, "type": "quantitative"},
+                    "y": {"field": y_field, "type": "quantitative"},
+                    "color": {"value": "blue"}
+                }
+            },
+            {
+                "mark": "line",
+                "data": {"values": regression_df.to_dict(orient='records')},
+                "encoding": {
+                    "x": {"field": x_field, "type": "quantitative"},
+                    "y": {"field": y_field, "type": "quantitative"},
+                    "color": {"value": "red"}
+                }
+            },
+        ]
+    }
+
+    base_name = os.path.splitext(csv_file)[0]
+    file_path = f"{sanitize_filename(classification_col_types)}_{name}.vega.json"
+    with open(file_path, 'w') as file:
+        json.dump(vega_lite_config, file, indent=4)
+
+
+def generate_vega_lite_json_diffpercent(
+        csv_file,
+        classification_col,
+        classification_col_types,
+        x_field,
+        y_field,
+        fields,
+        nominal_fields,
+        name,
+        include_all_data=False):
+
+    tooltip_config = [{"field": field,
+                       "type": "nominal" if field in nominal_fields else "quantitative",
+                       "title": field} for field in fields]
+    vega_lite_config = {
+        "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
+        "description": "A scatterplot",
+        "data": {
+            "url": csv_file},
+        "transform": [] if include_all_data else [
+            {
+                "filter": f"datum['{classification_col}'] == '{classification_col_types}'"}],
+        "layer": [
+            {
+                "mark": {
+                    "type": "rule",
+                            "color": "grey",
+                            "strokeWidth": 3},
+                "encoding": {
+                    "y": {
+                        "datum": 0}}},
+            {
+                "mark": "point",
+                "encoding": {
+                    "x": {
+                        "field": x_field,
+                        "type": "quantitative"},
+                    "y": {
+                        "field": y_field,
+                        "type": "quantitative"},
+                    "tooltip": tooltip_config}}]}
+
+    base_name = os.path.splitext(csv_file)[0]

```

Comment:
I'm a little confused by this variable: it's never used? An IDE+linter would be able to catch that this is a never-used variable